python/ray/dag/dag_node_operation.py

```diff
@@ -367,4 +407,56 @@ def _generate_actor_to_execution_schedule(
                     )
     for _, candidates in actor_to_candidates.items():
         assert len(candidates) == 0
+    for actor_handle, execution_schedule in actor_to_execution_schedule.items():
+        print(f"Actor {actor_handle._ray_actor_id} schedule: {execution_schedule}")
     return actor_to_execution_schedule
+
+
+def _optimize_execution_schedule(
```

Comment:
can you add unit tests like @kevin85421 did before? (not e2e, but unit level testing)

Reply:
yes, will do.

Reply:
Done.